### PR TITLE
applications: asset_tracker_v2: Use CONFIG_LTE_NEIGHBOR_CELLS_MAX

### PR DESCRIPTION
**Pull request by trantanen**
- Instead of using hard coded value 17, which is maximum for CONFIG_LTE_NEIGHBOR_CELLS_MAX, we should just use CONFIG_LTE_NEIGHBOR_CELLS_MAX as the size for the neighbor cell table.
- Adding native_posix compilations into sample.yaml.

### CODE REVIEW

**Review by jtguggedal — APPROVED** [review]
LGTM

**Review by tokangas — APPROVED** [review]

**Comment by simensrostad** [comment]
@trantanen Its the on commit tests that fail for the application. This must be handled in firmware and can be reproduced by building the application for Native Posix.

**Comment by trantanen** [comment]
> @trantanen Its the on commit tests that fail for the application. This must be handled in firmware and can be reproduced by building the application for Native Posix.

I've done updates based on what has been found out. See PR description for more information.

File: applications/asset_tracker_v2/sample.yaml [hunk 1]
```diff
@@ -4,10 +4,12 @@ tests:
   applications.asset_tracker_v2.nrf_cloud:
     build_only: true
     build_on_all: true
-    platform_allow: nrf9160dk_nrf9160_ns thingy91_nrf9160_ns native_posix
+    platform_allow: nrf9160dk_nrf9160_ns thingy91_nrf9160_ns native_posix qemu_x86
     integration_platforms:
       - nrf9160dk_nrf9160_ns
       - thingy91_nrf9160_ns
+      - native_posix
+      - qemu_x86
```
Comment (trantanen) [inline applications/asset_tracker_v2/sample.yaml:12]: Do we want to build for qemu_x86 too?

Comment (simensrostad) [reply applications/asset_tracker_v2/sample.yaml:12]: Since we are removing support for QEMU I thinkn we can remove it

Comment (trantanen) [reply applications/asset_tracker_v2/sample.yaml:12]: Yep, and seems it doesn't work:
```
CMake Error at /jenkins_workspace/workspace/latest_sdk-nrf_PR-10326/zephyr/cmake/compiler/gcc/target.cmake:43 (message):
  C compiler
  /home/ncs/toolchains/fcef6322b4224bba80652b6ea86eebc963247c5d/opt/zephyr-sdk/x86_64-zephyr-elf/bin/x86_64-zephyr-elf-gcc
  not found - Please check your toolchain installation
Call Stack (most recent call first):
  /jenkins_workspace/workspace/latest_sdk-nrf_PR-10326/zephyr/cmake/modules/FindTargetTools.cmake:102 (include)
  /jenkins_workspace/workspace/latest_sdk-nrf_PR-10326/zephyr/cmake/modules/kernel.cmake:25 (find_package)
  /jenkins_workspace/workspace/latest_sdk-nrf_PR-10326/zephyr/cmake/modules/zephyr_default.cmake:117 (include)
  /jenkins_workspace/workspace/latest_sdk-nrf_PR-10326/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:66 (include)
  /jenkins_workspace/workspace/latest_sdk-nrf_PR-10326/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:92 (include_boilerplate)
  CMakeLists.txt:9 (find_package)

-- Configuring incomplete, errors occurred!
```

**Review by simensrostad — APPROVED** [review]

**Pull request merged**